doc/grammars/type.abnf

```diff
@@ -60,11 +60,23 @@ Array
 	= 1*(TokenSquareBracketOpen TokenSquareBracketClose)
 
 ArrayShape
-    = TokenCurlyBracketOpen ArrayShapeItem *(TokenComma ArrayShapeItem) TokenCurlyBracketClose
+	= Shape
 
 ArrayShapeItem
-    = (ConstantString / ConstantInt / TokenIdentifier) TokenNullable TokenColon Type
-    / Type
+	= ShapeItem
+
+Shape
+	= TokenCurlyBracketOpen ShapeItem *(TokenComma ShapeItem) TokenCurlyBracketClose
+
+ShapeItem
+	= (ConstantString / ConstantInt / TokenIdentifier) TokenNullable TokenColon Type
+	/ Type
```

Comment:
As mentioned in the comment, this only makes sense for array shapes. For object shapes you need sth like


~~~abnf
ObjectShapeItem = TokenIdentifier [TokenNullable] TokenColon Type
~~~